Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -4618,9 +4598,14 @@ PartialResult WARN_UNUSED_RETURN BBQJIT::addCallIndirect(unsigned tableIndex, co
 
             auto indexEqual = m_jit.branchPtr(CCallHelpers::Equal, calleeSignatureIndexTmp, TrustedImmPtr(TypeInformation::get(originalSignature)));
 
-            if (needsSubtypeCheck)
-                addRTTSlowPathJump(originalSignature.index(), calleeRTT);
-            else
+            if (needsSubtypeCheck) {
+                auto jump = m_jit.jump();
+                auto returnLabel = m_jit.label();
+                m_latePaths.append([jump, returnLabel, signature = originalSignature.index(), calleeRTT](BBQJIT& bbq, CCallHelpers& jit) {
+                    jump.link(jit);
+                    bbq.emitSlowPathRTTCheck(returnLabel, signature, calleeRTT);
+                });
```

Comment:
This doesn't seem like something that should be a late path. It could happen 100% of the time.

Reply:
Yeah, removed the late path.